components/director/internal/tenantmapping/cert_service_context_provider.go

```diff
@@ -49,39 +50,74 @@ func (m *certServiceContextProvider) GetObjectContext(ctx context.Context, reqDa
 	// This if is needed to separate the director from ord flow because for the director flow we need to use the internal ID of the subaccount
 	// whereas in the ord flow we expect external IDs in order ord views to work properly(using Automatic Scenario Assignments)
 	log.C(ctx).Infof("Matched component name is %s", matchedComponentName[0])
-	if matchedComponentName[0] == "director" { // Director Flow, do the tenant conversion
-		scopes := "runtime:read runtime:write tenant:read"
 
-		log.C(ctx).Infof("Getting the tenant with external ID: %s", externalTenantID)
-		tenantMapping, err := m.tenantRepo.GetByExternalTenant(ctx, externalTenantID)
-		if err != nil {
-			if apperrors.IsNotFoundError(err) {
-				log.C(ctx).Warningf("Could not find tenant with external ID: %s, error: %s", externalTenantID, err.Error())
+	// the authID in this flow is an OU selected by the Connector
+	externalTenantID := authDetails.AuthID
+
+	if matchedComponentName[0] != "director" { // ORD Flow, set the external tenant ID both for internal and external tenants
+		objCtx := NewObjectContext(NewTenantContext(externalTenantID, externalTenantID), p.tenantKeys, "", authDetails.Region, "", authDetails.AuthID, authDetails.AuthFlow, consumer.Runtime, CertServiceObjectContextProvider)
+		log.C(ctx).Infof("Successfully got object context: %+v", objCtx)
+		return objCtx, nil
+	}
 
-				log.C(ctx).Infof("Returning tenant context with empty internal tenant ID and external ID %s", externalTenantID)
-				return NewObjectContext(NewTenantContext(externalTenantID, ""), m.tenantKeys, scopes, authDetails.Region, "", authDetails.AuthID, authDetails.AuthFlow, consumer.Runtime, CertServiceObjectContextProvider), nil
-			}
-			return ObjectContext{}, errors.Wrapf(err, "while getting external tenant mapping [ExternalTenantID=%s]", externalTenantID)
+	extraData := reqData.GetExtraDataWithDefaults()
+
+	logger := log.C(ctx).WithFields(logrus.Fields{
+		"consumer_type": extraData.ConsumerType,
+	})
+	ctx = log.ContextWithLogger(ctx, logger)
+
+	if extraData.AccessLevel != "" {
+		var err error
+		externalTenantID, err = reqData.GetExternalTenantID() // will return tenant ID from header if it exists
+		if err != nil {
+			log.C(ctx).WithError(err).Errorf("Failed to get external tenant ID: %v", err)
+			return ObjectContext{}, err
 		}
+	}
 
-		objCtx := NewObjectContext(NewTenantContext(externalTenantID, tenantMapping.ID), m.tenantKeys, scopes, authDetails.Region, "", authDetails.AuthID, authDetails.AuthFlow, consumer.Runtime, CertServiceObjectContextProvider)
+	scopes, err := p.directorScopes(reqData.GetConsumerTypeFromExtra())
```

Comment:
We already have the conusmer type in extradata.ConsumerType. The way it was populated there makes sure that the consumer type is never empty (it will be runtime if empty initially). Getting it with this method here might return an empty string while the one above will not be. Is that desired? If not, use extradata.ConsumerType instead of calling this method.

Reply:
we do this in order to get the `default` scopes, instead of the runtime ones